compiler/parser/chapel.lex

```diff
@@ -91,8 +91,9 @@ ident            {letter}({letter}|{digit}|"$")*
 
 binaryLiteral    0[bB]{bit}+
 octalLiteral     0[oO]{octDigit}+
+decimalLiteral   {digit}(_|{digit})*
 hexLiteral       0[xX]{hexDigit}+
-intLiteral       {digit}+|{binaryLiteral}|{octalLiteral}|{hexLiteral}
+intLiteral       {binaryLiteral}|{octalLiteral}|{decimalLiteral}|{hexLiteral}
```

Comment:
I think this lexer change is missing the corresponding change for floating point literals.

Reply:
I've now added support for binary, octal, hexadecimal, and floating point literals.